core/tests/test_utils.py

```diff
@@ -3702,7 +3708,12 @@ def save_new_topic(
             feconf.CURRENT_SUBTOPIC_SCHEMA_VERSION, next_subtopic_id,
             language_code, 0, feconf.CURRENT_STORY_REFERENCE_SCHEMA_VERSION,
             meta_tag_content, practice_tab_is_displayed,
-            page_title_fragment_for_web, skill_ids_for_diagnostic_test)
+            page_title_fragment_for_web, skill_ids_for_diagnostic_test,
+            story_exploration_mapping or {
+                story_id: []
+                for story_id
+                in (canonical_story_ids or []) + (additional_story_ids or [])
+            })
```

Comment:
Can we expect a valid topic to have story_id in `story_exploration_mapping` which is neither in canonical_story_ids or additional_story_ids?

Reply:
@DubeySandeep I confirm that is not possible, a story has to be in one or the other of these two lists.

Reply:
In general, are we allowing users to edit `story_exploration_mapping` or is it generated using some fields? If the latter one, I would suggest generating it in the utils functions as well instead of allowing users to configure it.

Reply:
@DubeySandeep I initially wanted to generate it based on the stories the user gives it. However, if I wanted to get the appropriate exploration IDs, I would have to fetch each story, which must be saved to the datastore *before* the method is called. So my idea at the time was to default to a dictionary with all the story IDs mapped to empty lists of exploration IDs, but allow the user to provide a mapping when it should be initialized with exploration IDs.

Although now thinking back on the test setups I read, most of the topics were saved before the stories were saved, followed by linking the story with appropriate exploration IDs. So maybe I should just remove the argument altogether from the `save_new_topic` method.

Reply:
Actually, I found an [instance](https://github.com/oppia/oppia/blob/2160e4f73f725614d49580d628844765f2da5f73/core/controllers/story_viewer_test.py#L169) where in the test setup, the topic is saved after the story was linked explorations and saved. So should I go through with removing the parameter from `save_new_topic` method and enforce the rule that all test setups should save the topic before saving the story?

Reply:
@DubeySandeep Sorry I forgot to at-mention you in my above comment.

Reply:
@StephenYu2018 I think in general the tests should be set up following the user flow. So if a user wouldn't manually set that argument then don't include it as a special customization in the test construction method. Instead, the sequence of actions in the tests should follow the sequence of actions a user would use in practice, and if the field gets generated automatically in real-life usage then it should also get generated automatically when the tests are creating/saving the object. Does that help?

Reply:
This thread is outdated, as the relevant changes were made when I was trying to add the new field to `TopicModel` as opposed to adding the field to `TopicSummaryModel`, which is what I'm trying to do now.